app/colorCodeConversion.js

```diff
@@ -0,0 +1,45 @@
+const ColorPair = require("../app/ColorPair");
+const {MAJOR_COLOR_NAMES, MINOR_COLOR_NAMES} = require("../app/constants");
+
+module.exports = {
+    getColorFromPairNumber: function(pairNumber) {
+        let minorSize = MAJOR_COLOR_NAMES.length;
+    let majorSize = MINOR_COLOR_NAMES.length;
+
+    if (pairNumber < 1 || pairNumber > minorSize * majorSize) {
+        throw `Argument PairNumber:${pairNumber} is outside the allowed range`
+    }
+    let zeroBasedPairNumber = pairNumber - 1;
+    let majorIndex = parseInt(zeroBasedPairNumber / minorSize);
+    let minorIndex = parseInt(zeroBasedPairNumber % minorSize);
+    let pair = new ColorPair();
+    pair.majorColor = MAJOR_COLOR_NAMES[majorIndex];
+    pair.minorColor = MINOR_COLOR_NAMES[minorIndex];
+    return pair;
+    },
+
+    getPairNumberFromColor: function(pair) {
+        let majorIndex = -1;
+    for (let i = 0; i < MAJOR_COLOR_NAMES.length; i++) {
+        if (MAJOR_COLOR_NAMES[i] == pair.majorColor) {
+            majorIndex = i;
+            break;
+        }
+    }
+
+    let minorIndex = -1;

```

Comment:
Notice the repeated pattern between major and minor: Initialize to -1, iterate and match. Separate that into a function and call in both places to remove the duplication.

Reply:
please review all the comments